[Grid] Improve Grid2 upgrade experience (#45304)
mui/material-ui: docs/data/material/components/grid2/grid2.md

## Limitations

### Column direction

Using `direction="column"` or `direction="column-reverse"` is not supported.
The Grid component is designed to divide the container in columns, not rows.

> We advice against using the Grid component for vertical layouts.

Reviewer: ```suggestion
You should not use the Grid component on its own to stack layout elements vertically.
```

> Instead, you can use the [Stack component](/material-ui/react-stack/) inside your grid to create vertical layouts, as follows:

Reviewer: ```suggestion
Instead, you should use the [Stack component](/material-ui/react-stack/) inside of a Grid to create vertical layouts as shown below:
```
